File: leetcode/reverse_linked_name/1.js
function ListNode(val) {
    this.val = val
    this.next = null
}

var reverseList = function(head) {
    if(!head || !head.next)  return  head // 空链表或者只有头节点
    let cur = head //当前节点
    let pre = null
    while (cur) { //遍历
        const next = cur.next //后继节点
        cur.next = pre; //反过来, 将当前节点得指针指向前一个节点
        //如果cur 是头节点
        pre = cur; //前驱节点变成当前节点
        cur = next; // 后继节点变成当前节点
    }
    return cur
}
const n1 = new ListNode(1)
const n2 = new ListNode(2)
const n3 = new ListNode(3)
const n4 = new ListNode(4)
const n5 = new ListNode(5)
n1.next = n2
n2.next = n3
n3.next = n4
n4.next = n5
console.log(n1)

File: leetcode/reverse_linked_name/1.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if not head: return head
        prev = None
        cur = head
        while cur:
            cur.next, prev, cur = prev, cur, cur.next
        return prev
